docs/resources/publish_pipeline.md

```diff
@@ -0,0 +1,131 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "mezmo_publish_pipeline Resource - terraform-provider-mezmo"
+subcategory: ""
+description: |-
+  This resource will monitor a pipeline for changes, and publish it when necessary.
+  Configuration
+  To make sure a pipeline and its components exist before publishing, the configuration of this resource requires the use of child modules and depends_on. The pipeline's configuration should exist in a child module with an output of the pipeline's id and updated_at fields. This resource will then reference those fields, and be able to publish as needed when the pipeline changes.
+  The output can be done however the user chooses, as long as the id and timestamp are accessible in the root module. In other words, output can be an object of the entire pipeline, or individual fields for clearer naming.
+  Example
+  Child module in ./pipeline/main.tf
+  ```hcl
+  terraform {
+      requiredproviders {
+          mezmo = {
+              source = "registry.terraform.io/mezmo/mezmo"
+          }
+      }
+  }
+  output "mypipeline" {
+      value = mezmopipeline.mypipeline
+  }
+  resource "mezmopipeline" "mypipeline" {
+      title = "A pipeline to publish"
+  }
+  ... other sources, processors, destinations, etc.
+  ```
+  Root module in main.tf
+  ```hcl
+  terraform {
+      requiredproviders {
+          mezmo = {
+              source = "registry.terraform.io/mezmo/mezmo"
+          }
+      }
+  }
+  provider "mezmo" {
+      authkey = "my secret"
+  }
+  resource "mezmopublishpipeline" "publisher" {
```

Comment:
Should this be `resource "mezmo_publish_pipeline" "publisher"`. Looks like the resource name is missing the '_'s.

Reply:
Yes, this is super frustrating.  There's a bug in `go generate` such that it gives varies results.  One of the variations does this splitting.  Another removes some markdown header hashes (`##`), and yet another destroys the templating for alerts.  I can't seem to get one version to satisfy all things, so how we get around this is a bit unclear.

We might have to fix the docs manually and hope for the best later.

Reply:
Begrudgingly, I fixed these by hand.  I'm not sure why it wan't to remove underscores or hash marks from the template, but at least the main doc is correct.

Since the doc generation also includes a `Usage` (which pulls the example), the description was cleaned up so that they merge together.  The generation is not pulling the example part that contains the child module.  Duct tape and match sticks, I swear.